ds701_book/26-TimeSeries.qmd

```diff
@@ -0,0 +1,1114 @@
+
+
+## Seasonal ARIMA (SARIMA) models
+
+A type of time series model that extends ARIMA to support seasonality. The model
+is defined as:
+
+$$
+y_t = c + \phi_1 y_{t-1} + \phi_2 y_{t-2} + \ldots + \phi_p y_{t-p} + \epsilon_t + \theta_1 \epsilon_{t-1} + \theta_2 \epsilon_{t-2} + \ldots + \theta_q \epsilon_{t-q} + \Phi_1 Y_{t-s} + \Phi_2 Y_{t-2s} + \ldots + \Phi_P Y_{t-Ps} + \Theta_1 E_{t-s} + \Theta_2 E_{t-2s} + \ldots + \Theta_Q E_{t-Qs}
+$$
+
+where:
+
+- $y_t$ is the current value,
+- $c$ is a constant,
+- $\phi_1, \phi_2, \ldots, \phi_p$ are the AR coefficients,
+- $\epsilon_t$ is the white noise error term,
+- $\theta_1, \theta_2, \ldots, \theta_q$ are the MA coefficients,
+- $\Phi_1, \Phi_2, \ldots, \Phi_P$ are the seasonal AR coefficients,
+- $Y_{t-s}, Y_{t-2s}, \ldots, Y_{t-Ps}$ are the seasonal lagged observations,
+- $\Theta_1, \Theta_2, \ldots, \Theta_Q$ are the seasonal MA coefficients,
+- $E_{t-s}, E_{t-2s}, \ldots, E_{t-Qs}$ are the seasonal lagged forecast errors.
+
+**Key Points**:
+
+1. **Order (p, d, q) x (P, D, Q, s)**: The parameters of the SARIMA model where $p, d, q$ are the non-seasonal parameters, $P, D, Q$ are the seasonal parameters, and $s$ is the length of the seasonal cycle.
+2. **Stationarity**: Differencing is used to make the series stationary.
+3. **Parameter Estimation**: Methods like Maximum Likelihood Estimation (MLE) are used to estimate the parameters.
+
+
+## ARIMA/SARIMA Example
+
+Let's look at a concrete example of ARIMA/SARIMA using the Air Passengers dataset.
+
+```{python}
+import pandas as pd
+import numpy as np
+from statsmodels.tsa.seasonal import seasonal_decompose
+from statsmodels.tsa.statespace.sarimax import SARIMAX
+import matplotlib.pyplot as plt
+```
+
+We'll load the dataset and show the head.
+
+```{python}
+path = 'air_passengers_1949_1960.csv'
+data = pd.read_csv(path)
+data.head()
+```
+
+---
+
+Next, use pandas `date_range` method to create a new month column. We could have
+used date string conversion tools on the `Date` column alternatively.
+
+```{python}
+data['Month'] = pd.date_range(start='1949-01', periods=len(data), freq='ME')
+data.set_index('Month', inplace=True)
+
+data.head()
+```
+
+---
+
+As a next step, we'll log transform the data to stabilize the variance.
+
+```{python}
+# Log transform to stabilize variance
+data['Log_Passengers'] = np.log(data['Number of Passengers'])
+data.head()
+```
+
+---
+
+And finally, we'll perform a **classical additive seasonal decomposition** of the data as a point
+of comparison.
+
+```{python}
+# Seasonal decomposition
+decomposition = seasonal_decompose(data['Log_Passengers'], model='additive')
+decomposition.plot()
+plt.show()
+```
+
+## SARIMAX Model
+
+Now let's create an instance of a **SARIMAX** model -- <b>S</b>easonal 
+<b>A</b>utoregressive <b>I</b>ntegrated <b>M</b>oving <b>A</b>verage with 
+e<b>X</b>ogenous regressors.
+
+It extends the SARIMA model to support exogenous variables.
```

Comment:
What exogenous variables are used in this example?

Reply:
good point. i mention that we are not using any. That's the only model in statsmodels but default is `exog=None`.